spring-core/src/main/java/org/springframework/core/io/buffer/DataBufferUtils.java

```diff
@@ -622,7 +622,7 @@ public static Matcher matcher(byte[] delimiter) {
 	 * @since 5.2
 	 */
 	public static Matcher matcher(byte[]... delimiters) {
-		Assert.isTrue(delimiters.length > 0, "Delimiters must not be empty");
+		Assert.notEmpty(delimiters, "Delimiters must not be empty");
```

Comment:
We rely on nullability constraints in such cases, and `delimiters` should never be `null`, since `@Nullable` is not declared for that method parameter.

If a user supplies a `null` value for the var-args array, the resulting `NullPointerException` should suffice to inform them of their error.